src/LightSpeed/userFunction.js

```diff
@@ -53,33 +53,100 @@ const COMMON_SCOPE = {
     },
 };
 
-const COMMON_SPAWN_SCOPE = {
-    // Keep this in sync with SpawnScope class in
-    // hack-toy-apps/com.endlessm.LightSpeed/userScope.js
-    ticksSinceSpawn: 0,
+// Set method that checks type for number properties of objects
+function propNumFactory(obj, propName, valName) {
+    Object.defineProperty(obj, propName, {
+        set(newVal) {
+            if (Number.isNaN(Number(newVal)))
+                throw new TypeError(`${propName} must be a number.`);
+            else if (Number(newVal) < 0)
+                throw new RangeError(`${propName} must be a positive number.`);
+            valName = newVal; // eslint-disable-line no-param-reassign
+        },
+        get() {
+            return valName;
+        },
+    });
+}
+
+const NUMTYPE_HANDLER = {
+    set(obj, prop, value) {
+        if (prop in obj) {
+            if (Number.isNaN(Number(value)))
+                throw new TypeError(`Value for ${prop} must be a number.`);
+            else
+                return Reflect.set(obj, prop, value);
+        } else {
+            throw new TypeError(`Unknown property ${prop}.`);
```

Comment:
I'm hesitating about whether we do want to throw here or silently allow it as a normal Javascript object would. My understanding from reading the code is that this will mark on the third line of the following script as an error, is that correct?
```js
enemy.position.x = 3;
enemy.position.y = 4;
enemy.position.z = 5;  // !
```

I think it's probably a net positive, and it's OK to keep it for now and see how it turns out, although this seems like something that Chaim might feel strongly one way or the other on.

Reply:
Yes, that's correct. I added it because I thought it was in keeping with the types of errors we give for the scope properties in Sidetrack. I consider this part of the code to be potentially hard for kids to follow so I thought it would be better to opt for feedback on meeting the lesson goals rather than flexibility in this case. If it turns out to more confusing then I'm totally fine with removing it.